Add allocate_treatments function

Dimitry-Wintermantel opened the pull request:
As previously agreed I now provide here the allocate_treatment function that enables balanced treatment allocation based on specified covariates. The function ensures that treatment groups are created while minimizing within-group variance and allowing constraints such as equal replicates, standardized covariates, and stratified matching within subsets.

Dimitry-Wintermantel committed:
Initial commit - adding allocate_treatments function that assigns subjects to treatment groups while balancing specified covariates

m-Py commented:
Hi Dimitry,

thanks for your PR. Sorry for my late reaction, I was quite busy when you submitted and then forgot to get back to you sooner. I have two followup questions: 

- Could you change the target of your PR? I just created a new branch `bee_allocation`, which you can use. It is (currently) identical to main, which is however reserved for released versions of `anticlust`. 
- Could you provide a reference for your function? As discussed, it makes sense to explain where this function is coming from and why it is in `anticlust`. I assume that a working title / manuscript in preparation reference would suffice now. We might even synchronize when the function is included in `anticlust` and when you submit / publish your paper. I would also add to the documentation (but this is something that I can do) that this function implements the anticlustering specification that you used in your paper.

Thanks,

Martin

Dimitry-Wintermantel commented:
Hi Martin,

Sorry, dass ich erst so spät antworte. Ich hatte ein paar Deadlines. Bin
dann noch krank geworden und habe es dann vergessen. Ich arbeite jetzt
wieder an dem Paper in dem die Funktion vorgestellt wird. Wir können das
gerne mit der Submission synchronisieren.

Viele Grüße
Dimitry

On Thu, Mar 6, 2025 at 10:35 AM Martin Papenberg ***@***.***>
wrote:

> Hi Dimitry,
>
> thanks for your PR. Sorry for my late reaction, I was quite busy when you
> submitted and then forgot to get back to you sooner. I have two followup
> questions:
>
>    - Could you change the target of your PR? I just created a new branch
>    bee_allocation, which you can use. It is (currently) identical to
>    main, which is however reserved for released versions of anticlust.
>    - Could you provide a reference for your function? As discussed, it
>    makes sense to explain where this function is coming from and why it is in
>    anticlust. I assume that a working title / manuscript in preparation
>    reference would suffice now. We might even synchronize when the function is
>    included in anticlust and when you submit / publish your paper. I
>    would also add to the documentation (but this is something that I can do)
>    that this function implements the anticlustering specification that you
>    used in your paper.
>
> Thanks,
>
> Martin
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/m-Py/anticlust/pull/68#issuecomment-2703302675>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIETJRDL4RDRI2CRBSCIPBL2TAJE5AVCNFSM6AAAAABX5G4R7SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBTGMYDENRXGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: m-Py]*m-Py* left a comment (m-Py/anticlust#68)
> <https://github.com/m-Py/anticlust/pull/68#issuecomment-2703302675>
>
> Hi Dimitry,
>
> thanks for your PR. Sorry for my late reaction, I was quite busy when you
> submitted and then forgot to get back to you sooner. I have two followup
> questions:
>
>    - Could you change the target of your PR? I just created a new branch
>    bee_allocation, which you can use. It is (currently) identical to
>    main, which is however reserved for released versions of anticlust.
>    - Could you provide a reference for your function? As discussed, it
>    makes sense to explain where this function is coming from and why it is in
>    anticlust. I assume that a working title / manuscript in preparation
>    reference would suffice now. We might even synchronize when the function is
>    included in anticlust and when you submit / publish your paper. I
>    would also add to the documentation (but this is something that I can do)
>    that this function implements the anticlustering specification that you
>    used in your paper.
>
> Thanks,
>
> Martin
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/m-Py/anticlust/pull/68#issuecomment-2703302675>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIETJRDL4RDRI2CRBSCIPBL2TAJE5AVCNFSM6AAAAABX5G4R7SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOMBTGMYDENRXGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

Dimitry-Wintermantel committed:
Update allocation function and add second function for multi-level designs

Dimitry-Wintermantel commented:
Hi Martin, 

Sorry for the delay. The article on the allocation functions is now for about a month under review. I have changed the PR target to bee_allocation and updated the function and added the second one that we spoke about long time ago (allocation in multi-level designs). Thanks for allowing me to contribute those functions to anticlust.  

Best wishes,
Dimitry
